trieste/models/interfaces.py

```diff
@@ -84,17 +85,7 @@ def predict_y(self, query_points: TensorType) -> tuple[TensorType, TensorType]:
             f"Model {self!r} does not support predicting observations, just the latent function"
         )
 
-    def get_observation_noise(self) -> TensorType:
-        """
-        Return the variance of observation noise.
-
-        Note that this is not supported by all models.
-
-        :return: The observation noise.
-        """
-        raise NotImplementedError(f"Model {self!r} does not provide scalar observation noise")
-
-    def reparam_sampler(self, num_samples: int) -> ReparametrizationSampler:
+    def reparam_sampler(self: T, num_samples: int) -> ReparametrizationSampler[T]:
```

Comment:
wht not separating these samplers into their own PM interfaces?

Reply:
PM? (also I think there's some useful refactoring work possible re samplers, but not as part of this PR)

Reply:
ProbabilisticModeling subclasses